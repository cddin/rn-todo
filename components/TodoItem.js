import { View, StyleSheet, Text, Pressable } from "react-native";

const TodoItem = ({ data, onPress, onLongPress }) => {
  return (
    <Pressable
      onPress={() => onPress(data.id)}
      onLongPress={() => onLongPress(data.id)}
    >
      <View style={styles.container}>
        <Text style={data?.isDone ? styles.done : styles.notDone}>
          {data.title}
        </Text>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 5,
    marginBottom: 5,
    borderColor: "#aeaeae",
    borderStyle: "solid",
    borderWidth: 1,
  },
  done: {
    color: "#aeaeae",
    textDecorationLine: "line-through",
    textDecorationStyle: "solid",
  },
  notDone: {
    color: "#000000",
  },
});

export default TodoItem;
