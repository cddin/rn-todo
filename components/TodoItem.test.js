import React from "react";
import renderer from "react-test-renderer";
import TodoItem from "./TodoItem";
import { fireEvent, render } from "@testing-library/react-native";

const mockToggleTodoItemFn = jest.fn();
const mockDeleteTodoItemFn = jest.fn();
const data = { title: "xxx", id: 123 };
const elemenToTest = (
  <TodoItem
    data={data}
    onPress={mockToggleTodoItemFn}
    onLongPress={mockDeleteTodoItemFn}
  />
);

describe('<TodoItem data={{ title: "xxx" }} />', () => {
  it("has 1 child", () => {
    const tree = renderer.create(elemenToTest).toJSON();
    expect(tree.children.length).toBe(1);
  });
  //
  it("renders correctly across screens", () => {
    const tree = renderer.create().toJSON();
    expect(tree).toMatchSnapshot();
  });
  //
  it("press to trigger toggle with data id 123", () => {
    const { getByText } = render(elemenToTest);
    fireEvent.press(getByText("xxx"));
    expect(mockToggleTodoItemFn).toBeCalledWith(123);
  });
  //
  it("long press to trigger delete with data id 123", () => {
    const { getByText } = render(elemenToTest);
    fireEvent.press(getByText("xxx"));
    fireEvent(getByText("xxx"), "onLongPress");
    expect(mockToggleTodoItemFn).toBeCalledWith(123);
  });
});
