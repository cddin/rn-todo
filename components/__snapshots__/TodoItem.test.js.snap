// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem data={{ title: "xxx" }} /> renders correctly across screens 1`] = `null`;
